File: server/controller/user/accout.go
package user

import (
	"app/repository/model"
	user "app/repository/module/user"
	"encoding/json"
	"net/http"

	"github.com/go-chi/jwtauth/v5"
)

func SignIn(w http.ResponseWriter, r *http.Request) {

	var userSignIn model.Orderer

	json.NewDecoder(r.Body).Decode(&userSignIn)

	isValid := user.CheckIsUserSignIn(userSignIn)

	if isValid == true {

		result, _ := json.Marshal(map[string]interface{}{
			"Warn": "Phone is exist",
		})

		w.Write(result)
	} else {

		user.AddUser(userSignIn)
		userAddContract := user.CheckIsUser(userSignIn)

		tokenAuth := jwtauth.New("HS256", []byte("token"), nil)

		_, tokenString, _ := tokenAuth.Encode(map[string]interface{}{
			"id":    userAddContract.Id,
			"name":  userAddContract.Name,
			"phone": userAddContract.Phone,
		})

		cookie := http.Cookie{
			Name:     "jwt",
			Value:    tokenString,
			Path:     "/user_post",
			HttpOnly: true,
			MaxAge:   60 * 60,
		}

		http.SetCookie(w, &cookie)

		result, _ := json.Marshal(map[string]interface{}{
			"Token": tokenString,
		})

		w.Write(result)
	}
}

func Login(w http.ResponseWriter, r *http.Request) {

	var userLogin model.Orderer

	json.NewDecoder(r.Body).Decode(&userLogin)

	isValid := user.CheckIsUser(userLogin)

	if isValid.Id != 0 {

		tokenAuth := jwtauth.New("HS256", []byte("token"), nil)

		_, tokenString, _ := tokenAuth.Encode(map[string]interface{}{
			"id":    isValid.Id,
			"name":  isValid.Name,
			"phone": isValid.Phone,
		})

		cookie := http.Cookie{
			Name:     "jwt",
			Value:    tokenString,
			Path:     "/user_post",
			HttpOnly: true,
			MaxAge:   60 * 60,
		}

		http.SetCookie(w, &cookie)

		result, _ := json.Marshal(map[string]interface{}{
			"Token": tokenString,
		})

		w.Write(result)
	} else {
		result, _ := json.Marshal(map[string]interface{}{
			"Text": "None Exsit",
		})

		w.Write(result)
	}

}

File: clientuser/src/redux/thunk/accout.js
import Cookies from "js-cookie";

export const login = (name, pass, phone) => async dispatch => {

    const user = {
        name,
        pass,
        phone
    };

    await fetch("http://localhost:3000/user/login", {
        method: "POST",
        headers: {
            "Content-Type": "application/json",
        },
        body: JSON.stringify(user)
    })
        .then(res => res.json())
        .then(res => {
            if (res.Text) {
                dispatch({
                    type: "is_valid_accout",
                    payload: res.Text,
                })
            } else {
                dispatch({
                    type: "is_valid_accout",
                    payload: "",
                })
                Cookies.set("user", res.Token, { expires: 1, path: "/" });
                window.location.href = "http://localhost:3001/"
            }
        })
}

export const signIn = (name, pass, phone) => async dispatch => {

    const user = {
        name,
        pass,
        phone,
    }

    await fetch("http://localhost:3000/user/signin", {
        method: "POST",
        headers: {
            "Content-Type": "application/json",
        },
        body: JSON.stringify(user),
    })
        .then(res => res.json())
        .then(res => {
            if (res.Warn) {
                dispatch({
                    type: "is_valid_accout",
                    payload: res.Warn,
                })
            } else {
                Cookies.set("user", res.Token, { expires: 1, path: "/" })
                window.location.href = "http://localhost:3001/"
            }
        })
}